sentry_sdk/integrations/spark/spark_worker.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import absolute_import
+
+import sys
+
+from sentry_sdk import configure_scope
+from sentry_sdk.hub import Hub
+from sentry_sdk.utils import (
+    capture_internal_exceptions,
+    exc_info_from_error,
+    single_exception_from_error_tuple,
+    walk_exception_chain,
+    event_hint_with_exc_info,
+)
+
+
+def _capture_exception(exc_info, hub):
+    from pyspark.taskcontext import TaskContext
+
+    client = hub.client
+
+    client_options = client.options
+    mechanism = {"type": "spark", "handled": False}
+
+    exc_info = exc_info_from_error(exc_info)
+
+    exc_type, exc_value, tb = exc_info
+    rv = []
+
+    for exc_type, exc_value, tb in walk_exception_chain(exc_info):
+        if exc_type not in (SystemExit, EOFError, ConnectionResetError):
+            rv.append(
+                single_exception_from_error_tuple(
+                    exc_type, exc_value, tb, client_options, mechanism
+                )
+            )
+
+    if rv:
+        rv.reverse()
+        hint = event_hint_with_exc_info(exc_info)
+        event = {"level": "error", "exception": {"values": rv}}
+
+        taskContext = TaskContext._getOrCreate()
+
+        with configure_scope() as scope:
```

Comment:
Can you do this not as part of exception catching, but on every invocation of main? This way errors captured by the user will also have this context data

Reply:
So in this case the TaskContext is only created when the `main` function of the worker is called.

https://github.com/apache/spark/blob/master/python/pyspark/worker.py#L502-L507

Not sure the best way to patch `main` to make sure user errors have this context data.

Reply:
This should work:

```
with configure_scope() as scope:
    @scope.add_event_processor
    def process_event(event, hint):
        context = TaskContext.get()
        event.setdefault("tags", {}).setdefault("stageId", context.stageId())
        ...
        return event
```

I.e. never use the `scope.set_tag`

Reply:
What is the best way to test that the tags have been set correctly?

Reply:
I would send an event and assert the tags have been set correctly. See other tests in `tests/integrations/flask`, for example.